src/Mod/MeshPart/App/AppMeshPartPy.cpp

```diff
@@ -134,8 +134,8 @@ class Module : public Py::ExtensionModule<Module>
             "    MinLength (required, float)\n"
             "    Fineness (required, integer)\n"
             "    SecondOrder (optional, integral boolean)\n"
-            "    Optimize (optional, integeral boolean)\n"
-            "    AllowQuad (optional, integeral boolean)\n"
+            "    Optimize (optional, integral boolean)\n"
+            "    AllowQuad (optional, integral boolean)\n"

```

Comment:
This is maybe "integer boolean", talking about a non-zero natural number, like 1, 2, 3, 4, 5, etc., to indicate `True`, and the zero indicates `False`.

Reply:
@wmayer care to weigh in?

Reply:
Yes, this is definitely an integer boolean.

In that same file, starting in line 476 you can see the implementation of the function `meshFromShape`. The specific code that uses the arguments `SecondOrder`, `Optimize`, and `AllowQuad` starts in line 579.
```cpp
#if defined (HAVE_NETGEN)
        static char* kwds_fineness[] = {"Shape", "Fineness", "SecondOrder", "Optimize", "AllowQuad",NULL};
        PyErr_Clear();
        int fineness=0, secondOrder=0, optimize=1, allowquad=0;
        if (PyArg_ParseTupleAndKeywords(args.ptr(), kwds.ptr(), "O!i|iii", kwds_fineness,
                                        &(Part::TopoShapePy::Type), &shape, &fineness,
                                        &secondOrder, &optimize, &allowquad)) {
            MeshPart::Mesher mesher(static_cast<Part::TopoShapePy*>(shape)->getTopoShapePtr()->getShape());
            mesher.setMethod(MeshPart::Mesher::Netgen);
            mesher.setFineness(fineness);
            mesher.setSecondOrder(secondOrder != 0);
            mesher.setOptimize(optimize != 0);
            mesher.setQuadAllowed(allowquad != 0);
            return Py::asObject(new Mesh::MeshPy(mesher.createMesh()));
        }
```
As you can see, they are defined as `int`. Moreover, the arguments are passed to the `mesher` methods using a test, `mesher.setSecondOrder(secondOrder != 0)`.

What this does is convert the integer value into a boolean value. If `secondOrder` is in fact `0`, the test will return `False`, which is what will be passed to the function. If `secondOrder` is any other integer, the test will return `True`, which will be passed to the function.

If we go into the `Mesher.h` source, we see that the methods `setSecondOrder` et al., are defined to accept a boolean.
```cpp
    void setSecondOrder(bool on)
    { secondOrder = on; }
    void setOptimize(bool on)
    { optimize = on;}
    void setQuadAllowed(bool on)
    { allowquad = on;}
```

Reply:
Nice breakdown! I added the revisions to 20a27a9